warehouse/clickhouse/clickhouse.go

```diff
@@ -279,6 +289,52 @@ func generateArgumentString(arg string, length int) string {
 	return strings.Join(args, ",")
 }
 
+func castStringToArray(data string, dataType string) interface{} {
+	switch dataType {
+	case "array(int)":
+		dataInt := make([]int64, 0)
+		json.Unmarshal([]byte(data), &dataInt)
+		return dataInt
+	case "array(float)":
+		dataFloat := make([]float64, 0)
+		json.Unmarshal([]byte(data), &dataFloat)
+		return dataFloat
+	case "array(string)":
+		dataInterface := make([]interface{}, 0)
+		json.Unmarshal([]byte(data), &dataInterface)
+		dataString := make([]string, 0)
+		for _, value := range dataInterface {
+			if _, ok := value.(string); ok {
+				dataString = append(dataString, value.(string))
+			} else {
+				bytes, _ := json.Marshal(value)
+				dataString = append(dataString, string(bytes))
+			}
+		}
+		return dataString
+	case "array(datetime)":
+		dataTime := make([]time.Time, 0)
```

Comment:
should we do like https://github.com/rudderlabs/rudder-transformer/blob/master/warehouse/index.js#L137 to keep it consistent

Reply:
Internally clickhouse will convert the time value into the format set in the clickhouse settings when saving the time value into table